Detectors/TPC/simulation/src/Digitizer.cxx

```diff
@@ -65,9 +66,9 @@ DigitContainer *Digitizer::Process(TClonesArray *points)
   static ElectronTransport electronTransport;
   static PadResponse padResponse;
 
-  const int MCEventID = mgr->GetEntryNr();
-
-  static std::array<float, mNShapedPoints> signalArray;
+  /// \todo the size of the array should be retrieved from ParameterElectronics::getNShapedPoints()
+  static std::vector<float> signalArray;
+  signalArray.resize(eleParam.getNShapedPoints());
```

Comment:
since you use `eleParam.getNShapedPoints()` also below in a loop, please define a variable for this:
`const int nShapedPoints = eleParam.getNShapedPoints()` and use it here and below.